lib/linebuffer.js

```diff
@@ -213,12 +237,23 @@ module.exports.setSocket = function(sock) {
 
         flushtimeout = setTimeout(exports.flush, config.FLUSH_INTERVAL);
     }
+
+    return req;
```

Comment:
I feel like maybe this part can be cleaned up a little. Because in line:168 you assign `req` from this function, but then you also return it which is assigned line:151 of `connection-manager.js`. If `req` is only ever assigned inside this file (or this function) is there a need to keep it inside a closure and declare it on line:12?

Also conceptually, here's what is happening if I'm summarizing this correctly.

There is a `setConnection` function method inside `linebuffer` that creates a request object based on input properties that is then returned to `connection-manager` to hold onto for requests that it makes?

I feel like the roles/responsibilities here are a little mixed up. I would expect `linebuffer` to be only responsible for things related to the "line buffer" and if there is a `setConnection` function, that it would take in perhaps the request object to hold onto rather than being the one to create the request object. And a `setter` function that actually returns something else seems strange to me, like perhaps a true/false return would make sense, but it's like if I'm setting something, I would not expect to get something back out of it. Perhaps if you did, then you may want to call it `createConnection`, in which case it still seems to make much more sense being a function of the "connection manager" rather than a function of the "line-buffer".

Reply:
@mikehu, I tried to completely move `linebuffer.setConnection` into `connection-manager.createConnection` but so many things depend on it - I mean, both `file-utilities` and `windows-utilities` call `linebuffer.addMessage` which calls `linebuffer.flush` which is using that `req` from `linebuffer` - using whole `config` (assuming `config` may contain `req` as `config.req`) as a parameter/input for `flush` and/or `addMessage` doesn't make sense from the first sight or does it?

Previously, there was `linebuffer.setSocket` which was assigning globally declared `socket` inside `linebuffer`.

Reply:
@mikehu, I think my recent commit is the solution to your request - created `conn` object inside `connection-manager` containing `req`, `queue`, `exclude_regex`, and `flushtimeout` information and getting passed into `file-utilities` and `windows-utilities` then into `linebuffer`.